### ISSUE TITLE
Support objectHashIgnoreUnknownHack flag from rpt2

### ISSUE THREAD

**Issue by arthurdenner** [jaredpalmer/tsdx#294]
### Current Behavior
I'm using tsdx to bundle [react-semantic-ui-datepickers](https://github.com/arthurdenner/react-semantic-ui-datepickers) and I can't [upgrade `rollup-plugin-copy`](https://github.com/arthurdenner/react-semantic-ui-datepickers/pull/64) to v3 because the build fails with the error `Unknown object type "asyncfunction"`. This is because we are not passing the flag `objectHashIgnoreUnknownHack` to `rpt2` which would solve the issue.

Can we add support for this flag on the `build` command or is that kind of risky? The `rpt2` documentation states: `Setting this option to true will make object-hash ignore unknowns, at the cost of not invalidating the cache if ignored elements are changed.`

### Desired Behavior
Build can be successful if the plugin gets the flag as `true`.

### Suggested Solution
Add an option to the `build` command - `--ignore-unknown-hack`? - that results in the `objectHashIgnoreUnknownHack` being true while creating the `rollup` config.

### Who does this impact? Who is this for?
`tsdx` users with custom `rollup` configs using plugins that use async/await, for example.

### Describe alternatives you've considered
N/A.

### Additional context
N/A.

**Comment by sw-yx** [comment]
i dont 100% understand. what is rpt2? why are we adding a flag just to pass a flag to a specific dependency others may not have? i definitely dont like the suggested solution but think there is a bigger solve here. please explain your rpt2 usage.

**Comment by arthurdenner** [comment]
Hi @sw-yx! Sorry for not being so clear at first. 

So, `rpt2` stands for `rollup-plugin-typescript2`, that `tsdx` is using again now (see #287). That's its internal name.

The plugin has a property called `objectHashIgnoreUnknownHack` which should be used when you want to compile a plugin that uses some features, e.g. async/await.

I'm extending `tsdx`'s default rollup config to use `rollup-plugin-copy`, but its latest version uses async/await, therefore, my building is failing because `rpt2` can't compile it.

I'm suggesting adding a flag to `tsdx build` that will be passed to `rpt2` as `objectHashIgnoreUnknownHack: true` or adding this flag as true by default on `tsdx`, but the second option might not be so good because `rpt2` advises you not to use the flag unless you really need - which is my use case :sweat_smile:.

Does this clarify the problem?

**Comment by sw-yx** [comment]
ah i see. thanks for explaining! i guess i dont know enough to be helpful. will leave to @medelman17 or @jaredpalmer

**Comment by DevanB** [comment]
I've ran into this issue today too: https://github.com/vladshcherbin/rollup-plugin-copy/issues/16#issuecomment-481632072

While using `rollup-plugin-copy` the `objectHashIgnoreUnknownHack` flag needs to be set. 

Here is some history on the `rollup-plugin-typescript2` and async/await that is requiring this flag: https://github.com/ezolenko/rollup-plugin-typescript2/issues/105

**Comment by arthurdenner** [comment]
Another thing to notice is that the `objectHashIgnoreUnknownHack` flag may lead to cache issues and we don't support the `clean` flag at the moment too, which could prevent any issues.

**Comment by arvinsim** [comment]
While I was searching for ways to import assets by using plugins like [rollup-plugin-url](https://github.com/rollup/rollup-plugin-url) and [rollup-plugin-smart-asset](https://github.com/sormy/rollup-plugin-smart-asset), I get this error too.

**Comment by arthurdenner** [comment]
@sw-yx, seems like there are more use cases and more plugins a solution regarding the flag.

Can I open a PR and we continue the discussion there?

**Comment by arthurdenner** [comment]
For those who need to bypass this restriction, a workaround for now is to override the plugin on your custom config, like this:

```javascript
config.plugins = config.plugins.map(plugin => {
  if (plugin && plugin.name === 'rpt2') {
    return rpts2({
      clean: true,
      objectHashIgnoreUnknownHack: true,
      tsconfig: opts.tsconfig,
      tsconfigDefaults: {
        compilerOptions: {
          sourceMap: true,
          declaration: true,
          jsx: 'react',
        },
      },
      tsconfigOverride: {
        compilerOptions: {
          target: 'esnext',
        },
      },
    });
  }

  return plugin;
});
```

**Comment by sw-yx** [comment]
yeah this is above my paygrade. thanks @arthurdenner for providing the fix. will leave to jared n co

**Comment by ambroseus** [comment]
@arthurdenner  @sw-yx 
+1 `clean: true,  objectHashIgnoreUnknownHack: true` also need to start using `closure-compiler` plugin 
https://github.com/jaredpalmer/tsdx/issues/358#issuecomment-566232991

**Comment by jaredpalmer** [comment]
Let's make PR's and add docs for each.

**Comment by agilgur5** [comment]
For those who haven't seen the references above, wanted to note here that I've created https://github.com/ezolenko/rollup-plugin-typescript2/pull/203 (after pushing for https://github.com/puleos/object-hash/pull/90 to be merged).

As is mentioned here and elsewhere, `objectHashIgnoreUnknownHack` can cause some cache issues, and `clean` basically disables the cache -- both of which are suboptimal. Hence my pushing for an actual fix in `object-hash` and PR to `rpts2`.

**Comment by arthurdenner** [comment]
I've been following your effort to get this done and thank you, @agilgur5! Really appreciate it!

**Comment by agilgur5** [comment]
Out-of-the-box support for async rollup plugins was just released in [v0.13.0](https://github.com/jaredpalmer/tsdx/releases/tag/v0.13.0)!
